Fails to build on recent targets with -mtune=generic

I package openblas for Arch AUR, and I'm getting reports from multiple users that it fails to build with the following error. 

```
cc -c -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -O2 -DMAX_STACK_ALLOC=2048 -fopenmp -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=12 -DMAX_PARALLEL_NUMBER=1 -DUSE_TLS -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.11\" -msse3 -mssse3 -msse4.1 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=strmm_iutncopy -DASMFNAME=strmm_iutncopy_ -DNAME=strmm_iutncopy_ -DCNAME=strmm_iutncopy -DCHAR_NAME=\"strmm_iutncopy_\" -DCHAR_CNAME=\"strmm_iutncopy\" -DNO_AFFINITY -I.. -UDOUBLE  -UCOMPLEX -Wno-uninitialized -UDOUBLE -UCOMPLEX -UOUTER -ULOWER -UUNIT generic/trmm_utcopy_8.c -o strmm_iutncopy.o
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/immintrin.h:51,
                 from ../kernel/x86_64/../arm/../simd/intrin.h:51,
                 from ../kernel/x86_64/../arm/sum.c:33:
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/avxintrin.h:517:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_extractf128_ps’: target specific option mismatch
  517 | _mm256_extractf128_ps (__m256 __X, const int __N)
      | ^~~~~~~~~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../arm/../simd/intrin.h:63,
                 from ../kernel/x86_64/../arm/sum.c:33:
../kernel/x86_64/../arm/../simd/intrin_avx.h:36:17: note: called from here
   36 |     __m128 hi = _mm256_extractf128_ps(sum_halves, 1);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/immintrin.h:51,
                 from ../kernel/x86_64/../arm/../simd/intrin.h:51,
                 from ../kernel/x86_64/../arm/sum.c:33:
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/avxintrin.h:1454:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_castps256_ps128’: target specific option mismatch
 1454 | _mm256_castps256_ps128 (__m256 __A)
      | ^~~~~~~~~~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../arm/../simd/intrin.h:63,
                 from ../kernel/x86_64/../arm/sum.c:33:
../kernel/x86_64/../arm/../simd/intrin_avx.h:35:17: note: called from here
   35 |     __m128 lo = _mm256_castps256_ps128(sum_halves);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/immintrin.h:51,
                 from ../kernel/x86_64/../arm/../simd/intrin.h:51,
                 from ../kernel/x86_64/../arm/sum.c:33:
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/avxintrin.h:270:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_hadd_ps’: target specific option mismatch
  270 | _mm256_hadd_ps (__m256 __X, __m256 __Y)
      | ^~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../arm/../simd/intrin.h:63,
                 from ../kernel/x86_64/../arm/sum.c:33:
../kernel/x86_64/../arm/../simd/intrin_avx.h:34:18: note: called from here
   34 |     sum_halves = _mm256_hadd_ps(sum_halves, sum_halves);
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/immintrin.h:51,
                 from ../kernel/x86_64/../arm/../simd/intrin.h:51,
                 from ../kernel/x86_64/../arm/sum.c:33:
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/avxintrin.h:270:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_hadd_ps’: target specific option mismatch
  270 | _mm256_hadd_ps (__m256 __X, __m256 __Y)
      | ^~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../arm/../simd/intrin.h:63,
                 from ../kernel/x86_64/../arm/sum.c:33:
../kernel/x86_64/../arm/../simd/intrin_avx.h:33:25: note: called from here
   33 |     __m256 sum_halves = _mm256_hadd_ps(a, a);
      |                         ^~~~~~~~~~~~~~~~~~~~
make[1]: *** [Makefile.L1:691: ssum_k.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/var/tmp/pamac-build-username/openblas-lapack/src/OpenBLAS-0.3.11/kernel'
make: *** [Makefile:165: libs] Error 1
```

One fix is to pass "-march=native" to the CFLAGS and it builds fine, this is for a zen2 cpu. I don't observe the issue on my machine with a zen1 cpu so it must be target dependent. By looking at the recent changelog, it doesn't look like it's been fixed since the 0.3.11 release.

--- a/Makefile.x86_64
+++ b/Makefile.x86_64
@@ -74,8 +74,10 @@ ifndef NO_AVX2
 ifeq ($(C_COMPILER), GCC)
 # AVX2 support was added in 4.7.0
 GCCVERSIONGTEQ4 := $(shell expr `$(CC) -dumpversion | cut -f1 -d.` \>= 4)
+GCCVERSIONGTEQ5 := $(shell expr `$(CC) -dumpversion | cut -f1 -d.` \>= 5)
 GCCMINORVERSIONGTEQ7 := $(shell expr `$(CC) -dumpversion | cut -f2 -d.` \>= 7)
-ifeq ($(GCCVERSIONGTEQ4)$(GCCMINORVERSIONGTEQ7), 11)
+GCCVERSIONCHECK := $(GCCVERSIONGTEQ5)$(GCCVERSIONGTEQ4)$(GCCMINORVERSIONGTEQ7)
+ifeq ($(GCCVERSIONCHECK), $(filter $(GCCVERSIONCHECK), 011 110 111))
 CCOMMON_OPT += -mavx2
 endif
 else 
@@ -86,8 +88,10 @@ endif
 ifeq ($(F_COMPILER), GFORTRAN)
 # AVX2 support was added in 4.7.0
 GCCVERSIONGTEQ4 := $(shell expr `$(FC) -dumpversion | cut -f1 -d.` \>= 4)
+GCCVERSIONGTEQ5 := $(shell expr `$(FC) -dumpversion | cut -f1 -d.` \>= 5)
 GCCMINORVERSIONGTEQ7 := $(shell expr `$(FC) -dumpversion | cut -f2 -d.` \>= 7)
-ifeq ($(GCCVERSIONGTEQ4)$(GCCMINORVERSIONGTEQ7), 11)
+GCCVERSIONCHECK := $(GCCVERSIONGTEQ5)$(GCCVERSIONGTEQ4)$(GCCVERSIONMINORGTEQ7)
+ifeq ($(GCCVERSIONCHECK), $(filter $(GCCVERSIONCHECK), 011 110 111))
 FCOMMON_OPT += -mavx2
 endif
 else

